config/settings/test.yml

```diff
@@ -1,3 +1,6 @@
+hostname: www.example.com # possible fallback for unsafe request.host
+virtual_hosts: ["www.example.com", "localhost"] # Safe host names
```

Comment:
`hostname` seems like what you want.  I'm not sure how it's used right now, but might it be renamed to `hostnames` and have the value be `["www.example.com", "localhost"]` to avoid duplication?

Reply:
We've always had that hostname one, that gets set at application initialization. You can take a look at production.rb, dev.rb, test.rb to see how it is used there (and elsewhere). It's important that that always be a single "default" host.

I do think that virtual_hosts tend to be a better way to communicate what a "safe" virtual host is. Those are the only virtual hosts we support, if its not in that list, we should raise an exception or fallback to that hostname.

These two are distinctly different.